contrib/vendor-bitcoin-core.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+set -e
+
+CORE_VENDOR_VERSION=""          # bitcoin repo tagged version to vendor.
+
+DEFAULT_DEPEND_DIR="depend"
+DEFAULT_CORE_REPO=https://github.com/bitcoin/bitcoin.git
+
+: "${CORE_VENDOR_DEPEND_DIR:=$DEFAULT_DEPEND_DIR}"
+: "${CORE_VENDOR_REPO:=$DEFAULT_CORE_REPO}"
+
+# CP_NOT_CLONE lets us just copy a directory rather than git cloning.
+# This is usually a bad idea, since it will bring in build artifacts or any other
+# junk from the source directory, but may be useful during development or CI.
+: "${CORE_VENDOR_CP_NOT_CLONE:=no}"
+
+function usage() {
+    echo ""
+    echo "Usage: script <bitcoin-core-version-tag>"
+    echo ""
+    echo -ne '\t'
+    echo "$0 v0.21.2"
+    echo ""
+}
+
+if $# < 1; then
+   usage
+fi
+
+echo "Using depend directory $CORE_VENDOR_DEPEND_DIR. Set CORE_VENDOR_DEPEND_DIR to override."
+echo "Using bitcoin repository $CORE_VENDOR_REPO. Set CORE_VENDOR_REPO to override."
+
+# Parse command-line options
+FORCE=no
+while (( "$#" )); do
+    case "$1" in
+    -f)
+        FORCE=yes
+        ;;
+    *)
+        if [ -z "$CORE_VENDOR_VERSION" ]; then
+            CORE_VENDOR_VERSION="$1"
+        else
+            echo "WARNING: ignoring unknown command-line argument $1"
+        fi
+        ;;
+    esac
+    shift
+done
+
+echo "Vendoring Bitcoin Core version: $CORE_VENDOR_VERSION"
```

Comment:
In 902bc9ca085d9ebe2699044f369b225bcf0842e2:

The rust-secp version of this allows you to leave the revision unset and just use whatever code is present in the directory you're copying. We should preserve this behavior.

Reply:
Cracking the whip on me, I was too lazy to parse out the version number using regex - I'll do it now :(

Reply:
Done, that wasn't so painful.